test/gtest/ucp/test_ucp_proto_select.cc

```diff
@@ -0,0 +1,447 @@
+/**
+ * Copyright (c) NVIDIA CORPORATION & AFFILIATES, 2024. ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#include "ucp_test.h"
+
+extern "C" {
+#include <ucp/core/ucp_ep.inl>
+#include <uct/base/uct_component.h>
+#include <uct/base/uct_iface.h>
+#include <uct/base/uct_md.h>
+#include <ucp/proto/proto_select.inl>
+#include <ucs/sys/topo/base/topo.h>
+}
+
+struct mock_t {
+public:
+    typedef void (*func_t)(void);
+    typedef std::map<func_t *, std::unique_ptr<mock_t>> mock_map_t;
+
+    ~mock_t()
+    {
+        *m_orig_ptr = m_orig;
+    }
+
+    static void cleanup()
+    {
+        g_mock_map.clear();
+    }
+
+    template <typename Fn>
+    static mock_t *setup(Fn *orig_ptr, Fn mock)
+    {
+        func_t *ptr = reinterpret_cast<func_t *>(orig_ptr);
+        auto it = g_mock_map.find(ptr);
+        if (it == g_mock_map.end()) {
+            mock_t * new_mock = new mock_t(ptr, reinterpret_cast<func_t>(mock));
```

Comment:
There's `make_unique` since c++14. Currently we use c++11.
Let's see if [this one](https://github.com/openucx/ucx/pull/10018) gets merged soon. If yes - can consider using `make_unique` here then.

Reply:
Ignore.